### PR TITLE
[CICD] Add CUDA to the platform configuration's general workflow.

### PR DESCRIPTION
**Pull request by AlexMa616**
## Summary

Enables end-to-end CUDA wheel builds and GPU integration testing for
PyTorch-Plugin-FL with PyTorch 2.10.0 + CUDA 13.0, FlagGems 5.3, and
FlagCX 0.13.0.

The main CI change is a chip-neutral common workflow architecture. The common
workflows contain no CUDA-specific branches: runner labels, container image,
mounts, environment variables, setup logic, and test commands are supplied by
the platform configuration.

For platforms using this pipeline, adding another accelerator only requires a
new `.github/configs/<platform>.yml` and corresponding `set_env_<platform>.sh`;
the common workflows do not need to change.

### What's included

**Platform-configured common CI**
- Adds reusable orchestration, wheel-build, and integration-test workflows:
  - `.github/workflows/all-tests-common.yml`
  - `.github/workflows/build-wheel-common.yml`
  - `.github/workflows/integration-tests-common.yml`
- Automatically discovers platform configurations under `.github/configs/`
  and runs them through the common pipeline.
- Validates each platform configuration before allocating its accelerator
  runner.
- Adds a generic integration-test runner that executes the configured commands
  sequentially and propagates the first failure.
- Keeps the existing MetaX build and integration workflows enabled.

**CUDA platform definition**
- Adds `.github/configs/cuda.yml` with:
  - NVIDIA self-hosted runner labels
  - CUDA CI image and container options
  - local Qwen3-0.6B model mount
  - device, operator, FlagGems runtime, factory, inference, and training tests
- Adds `.github/scripts/set_env_cuda.sh` to configure and validate:
  - PyTorch `2.10.0+cu130`
  - CUDA runtime `13.0`
  - FlagGems `5.3`
  - FlagCX `0.13.0`
  - CMake, CUDA, NCCL, FlagGems, and FlagCX library paths
- Converts the CUDA-specific build and integration workflows into thin wrappers
  around the common pipeline.

**Build and runtime compatibility**
- Builds wheels with `python -m build --wheel --no-isolation`, reusing the
  dependencies already provided by the CUDA image.
- Verifies the generated wheel by installing it and checking its metadata and
  packaged files.
- Makes artifact upload non-blocking so transient large-file upload failures do
  not hide successful builds and tests.
- Supports both `FlagGems::operators` and legacy `operators` CMake targets.
- Resolves the installed FlagGems library directory from the exported target.

**CUDA dispatch fixes**
- Routes missing backward, optimizer, sampling, and scalar operations to CUDA,
  including:
  - `add_.Tensor`
  - `_log_softmax` and its backward path
  - `_softmax_backward_data`
  - `div.Scalar`
  - `sort`, `topk`, and `multinomial`
  - required `_foreach_*` optimizer operations
- Keeps explicitly overridable FlagGems Python kernels out of the default route
  while allowing them in the configuration consistency checks.
- Prevents dispatch-only BMM tests from entering FlagGems autotuning and adds
  subprocess timeouts for BMM and cat dispatch probes.
- Loads Qwen models without requiring `accelerate`.

## Testing

Manual validation was performed on an NVIDIA DGX A100 node with 8 GPUs:

- CUDA image smoke test:
  - PyTorch `2.10.0+cu130`
  - CUDA runtime `13.0`
  - 8 × NVIDIA A100-SXM4-40GB detected
  - FlagGems `5.3.0rc2`
  - FlagCX `0.13.0`
  - CUDA tensor execution passed
- Wheel build, installation, and package verification passed.
- Qwen3-0.6B inference: **4 passed**.
- Qwen3-0.6B training: **3 passed**.
- Autograd regression tests for gradient accumulation: **3 passed**.
- FlagGems configuration consistency checks: **6 passed**.
- YAML parsing, workflow shell syntax, Ruff, and actionlint validation passed.
- Common integration runner ordering and failure propagation were verified.

The branch was rebased onto the latest upstream `main`. The complete GPU suite
has not been rerun after that rebase and will be exercised by the PR CI.

## Notes / follow-ups

- CUDA jobs require access to:
  - `harbor.baai.ac.cn/flagos-dev/pytorch-plugin-fl:manual-20260728-cuda-dev`
  - `/data/models/Qwen/Qwen3-0.6B` on the self-hosted runner
- Artifact upload remains best-effort because the self-hosted runner can
  encounter connection resets during large uploads. Build and test failures
  still fail the workflow normally.

### CODE REVIEW

File: torch_fl/backends.conf [hunk 1]
```diff
@@ -70,6 +77,22 @@ lt.Tensor = cuda
...
 
+# div
+div.Scalar = cuda
+
+# _foreach_* optimizer ops
+_foreach_mul_.Scalar = cuda
+_foreach_add_.Scalar = cuda
+_foreach_addcdiv_.ScalarList = cuda
+_foreach_addcmul_.Scalar = cuda
+_foreach_lerp_.Scalar = cuda
```
Comment (lvyufeng) [inline torch_fl/backends.conf:88]: config修改的原因是？

Comment (AlexMa616) [reply torch_fl/backends.conf:88]: Add operator routing for operators that are actually required by Autograd, Qwen3 inference, and training in CUDA integration tests but have not yet been registered, to prevent "backend not registered" runtime errors.

**Review by lvyufeng — APPROVED** [review]

**Review by zhaoyinglia — APPROVED** [review]
